provider/cmd/app/app.go

```diff
@@ -455,3 +464,37 @@ func runStreamingServer(ctx context.Context, setupLog, log logr.Logger, srv *ser
 	}
 	return nil
 }
+
+func runMetricsServer(ctx context.Context, setupLog logr.Logger, addr string) error {
+	setupLog.Info("Starting metrics server on " + addr)
+
+	mux := http.NewServeMux()
+	mux.Handle("/metrics", promhttp.Handler())
+
+	srv := http.Server{
+		Addr:         addr,
+		Handler:      mux,
+		ReadTimeout:  5 * time.Second,
+		WriteTimeout: 5 * time.Second,
+		IdleTimeout:  2 * time.Second,
+	}
+
+	go func() {
+		<-ctx.Done()
+		setupLog.Info("Shutting down streaming server")
+		shutdownCtx, cancel := context.WithTimeout(context.Background(), time.Second)
+		defer cancel()
+		locErr := srv.Shutdown(shutdownCtx)
+		if locErr != nil {
```

Comment:
optional: I would write line 487 & 488 as:
`if locErr := srv.Shutdown(shutdownCtx); locErr != nil {`

same for line 494 & 495

Reply:
i personally don't like this style. Juniors have problem understand life scope of variable properly (personally tested) and you can very easy do shadowing what is just more confusing.
+ i saw very long if statements with this style which are worse readable.
So i don't use this short format.

Reply:
I don't think so its worse readable. And plus point is variables declared by the statement are only in scope until the end of the if. Anyway it was optional.